src/psyclone/transformations.py

```diff
@@ -313,12 +313,171 @@ def apply(self, node, options=None):
         return schedule, keep
 
 
+class OMPTaskloopTrans(ParallelLoopTrans):
+
+    '''
+    Adds an OpenMP taskloop directive to a loop.
+
+    TODO: #1364 Taskloops do not yet support reduction clauses.
+
+    For example:
+
+    >>> from pysclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.transformations import OMPParallelTrans, OMPSingleTrans
+    >>> from psyclone.transformations import OMPTaskloopTrans
+    >>> singletrans = OMPSingleTrans()
+    >>> paralleltrans = OMPParallelTrans()
+    >>> tasklooptrans = OMPTaskloopTrans()
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    # Apply the OpenMP Taskloop transformation to *every* loop
+    # in the schedule.
+    # This ignores loop dependencies. These must be manually handled
+    # either through end of regions.
+    # TODO: #1368 These can also be handled through the taskwait directive
+    >>> for child in schedule.children:
+    >>>     tasklooptrans.apply(child)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # SINGLE region
+    >>> singletrans.apply(schedule.children)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # PARALLEL region
+    >>> paralleltrans.apply(schedule.children)
+    >>> schedule.view()
+
+    '''
+    def __init__(self, grainsize=None, num_tasks=None):
+        self._grainsize = None
+        self._num_tasks = None
+        self.omp_grainsize = grainsize
+        self.omp_num_tasks = num_tasks
+
+        super(OMPTaskloopTrans, self).__init__()
+
+    def __str__(self):
+        return "Adds an 'OpenMP TASKLOOP' directive to a loop"
+
+    @property
+    def omp_grainsize(self):
+        ''' 
+        Returns the grainsize that will be specified by
+        this transformation. By default the grainsize
+        clause is not applied, so grainsize is None.
+
+        :returns: The grainsize specified by this transformation
+        :rtype: int or None
+        '''
+        return self._grainsize
+
+    @omp_grainsize.setter
+    def omp_grainsize(self, value):
+        '''
+        Sets the grainsize that will be specified by
+        this transformation. Checks the grainsize is
+        a positive integer value or None.
+
+        :param value: Integer value to use in the grainsize clause.
+        :type value: int or None
+
+        :raises TransformationError: If value is not an int and is not None.
+        :raises TransformationError: If value is negative.
+        :raises TransformationError: if grainsize and num_tasks are \
```

Comment:
Start each param and raises text below with lower case. We chose this style because in the documentation this text goes after a hyphen symbol.

Reply:
Done